test/e2e_node/restart_test.go

```diff
@@ -118,5 +142,93 @@ var _ = framework.KubeDescribe("Restart [Serial] [Slow] [Disruptive]", func() {
 				By(fmt.Sprintf("Docker restart test passed with %d pods", len(postRestartRunningPods)))
 			})
 		})
+		Context("Live restore", func() {
+
+			It("should restore container when live restore is on", func() {
+				framework.SkipIfContainerRuntimeIs("rkt")
+
+				// Skip when docker engine version is less than 1.12
+				dockerVersion, err := getDockerVersion()
+				dockerRegex, _ := regexp.Compile(`1\.12\.[0-9]+`)
+				if !dockerRegex.Match([]byte(dockerVersion)) {
+					framework.Skipf("Live restore isn't suported before Docker Engine version 1.12")
+				}
+
+				// skip when docker live-restore is not enabled
+				dockerPid, err := ioutil.ReadFile("/var/run/docker.pid")
```

Comment:
i would prefer if we could have one standard helper for resolving the docker pid.

see: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/container_manager_linux.go#L635

could we refactor the above function some to have a shared utility for looking up the pids for docker in all use-cases?